app/src/main/java/org/p2p/wallet/infrastructure/network/provider/SeedPhraseProvider.kt

```diff
@@ -12,16 +12,36 @@ class SeedPhraseProvider(
 ) {
 
     val isAvailable: Boolean
-        get() = seedPhrase.isNotEmpty()
+        get() = getUserSeedPhrase().seedPhrase.isNotEmpty()
 
-    var seedPhrase: List<String> = emptyList()
+    fun getUserSeedPhrase(): UserSeedPhrase {
+        return UserSeedPhrase(
+            seedPhrase = seedPhrase,
+            provider = seedPhraseProvider
+        )
+    }
+
+    fun updateUserSeedPhrase(words: List<String>, provider: SeedPhraseProviderType) {
```

Comment:
setUserSeedPhrase, you replaced custom set, so it's better to name is like setter